src/repositories/GatewayRepository.ts

```diff
@@ -143,36 +125,44 @@ export class GatewayRepository { // extends Subject<Gateway> {
       },
       {
         ...initOptions,
-        modelName: 'gateway',
-        tableName: 'gateway',
+        modelName: 'Gateway',
+        tableName: 'gateways',
       },
     );
   }
 
   /**
-   * Creates a gateway model in the repository and syncs with database.
-   * @param {GatewayAttributes} gatewayAttributes
-   * @return {Promise<Gateway>}
+   * Saves a Gateway model in the repository.
+   * If a gateway does not exist, it creates, otherwise updates.
+   *
+   * @param gateway Gateway object to update.
+   *
+   * @returns Newly created or updated gateway object.
    */
-  public async create(gatewayAttributes: GatewayAttributes): Promise<Gateway> {
-    try {
-      const gateway: Gateway = await GatewayModel.create(gatewayAttributes) as Gateway;
-      this.format(gateway);
-      // this.newUpdate(gateway);
-      return gateway;
-    } catch (e) {
-      const errorContext = {
-        attributes: gatewayAttributes,
-        reason: e.message,
-      };
-      return Promise.reject(`Failed to create a gateway: ${JSON.stringify(errorContext)}`);
-    }
+  public async save(gateway: Gateway): Promise<Gateway> {
+    const definedOwnProps: string[] = Utils.getDefinedOwnProps(gateway);
+
+    await GatewayModel.upsert(
+      gateway,
+      {
+        fields: definedOwnProps,
+      },
+    );
+
+    const updatedGateway = await this.get(
+      gateway.gatewayAddress,
+    );
+
```

Comment:
Please, assert that it's not null (as .get() returns null or object).

Reply:
Done!